### PR TITLE
Fixed #28925 -- Made durations-only expressions work on SQLite and MySQL.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/28925

### CODE REVIEW

File: django/db/models/expressions.py [hunk 1]
```diff
@@ -404,28 +404,35 @@ def set_source_expressions(self, exprs):
...
-            return TemporalSubtraction(self.lhs, self.rhs).as_sql(compiler, connection)
+        lhs_type = lhs_output and lhs_output.get_internal_type()
+        rhs_type = rhs_output and rhs_output.get_internal_type()
+        datetime_fields = {'DateField', 'DateTimeField', 'TimeField'}
+        if not connection.features.has_native_duration_field and 'DurationField' in (lhs_type, rhs_type):
+            if not datetime_fields.isdisjoint((lhs_type, rhs_type)):
+                return DurationExpression(self.lhs, self.connector, self.rhs).as_sql(compiler, connection)
+            if isinstance(lhs, DurationValue):
+                value = lhs.value
+                lhs = Value(value.days * 86400000000 + value.seconds * 1000000 + value.microseconds)
```
Comment (charettes) [inline django/db/models/expressions.py]: Would `value.total_seconds() * 1_000_000` loose precision here?

**Comment by timgraham** [comment]
Sergey, did you notice the test failures?

**Comment by sir-sigurd** [comment]
@timgraham, yes, of course :slightly_smiling_face:. Still working on the fix.

**Comment by timgraham** [comment]
Please reopen if you return to it.

**Pull request closed without merging**